src/backend/EdFi.DataManagementService.Backend.Postgresql.Test.Integration/UpdateTests.cs

```diff
@@ -657,6 +659,94 @@ public void It_should_be_a_successful_update_for_2nd_transaction_due_to_retry()
+            IUpsertRequest sessionUpsertRequest = CreateUpsertRequest(
+                "Session",
+                _sessionDocumentUuid,
+                _sessionReferentialIdUuid,
+                """
+                {
+                    "sessionName": "Third Quarter"
+                }
+                """,
+                allowIdentityUpdates: true
+            );
+            var sessionUpsertResult = await CreateUpsert()
+                .Upsert(sessionUpsertRequest, Connection!, Transaction!, traceId);
+            sessionUpsertResult.Should().BeOfType<UpsertResult.InsertSuccess>();
+
+            IUpsertRequest courseOfferingUpsertRequest = CreateUpsertRequest(
+                "CourseOffering",
+                _courseOfferingDocumentUuid,
+                _courseOfferingReferentialIdUuid,
+                """
+                {
+                    "localCourseCode": "ABC",
+                    "sessionReference": {
+                        "sessionName": "Third Quarter"
+                    }
+                }
+                """,
+                CreateDocumentReferences(
+                    [new("CourseOffering", sessionUpsertRequest.DocumentInfo.ReferentialId.Value)]
+                )
+            );
+
+            var courseOfferingUpsertResult = await CreateUpsert()
+                .Upsert(courseOfferingUpsertRequest, Connection!, Transaction!, traceId);
+            courseOfferingUpsertResult.Should().BeOfType<UpsertResult.InsertSuccess>();
+        }
+
+        [Test]
+        public async Task Update_session()
+        {
+            var documentIdentityElement = new DocumentIdentityElement(
+                new JsonPath("$.sessionName"),
+                "Fourth Quarter"
+            );
+            IUpdateRequest sessionUpdateRequest = CreateUpdateRequest(
+                "Session",
+                _sessionDocumentUuid,
+                _sessionReferentialIdUuid,
+                """
+                {
+                    "sessionName": "Fourth Quarter"
+                }
+                """,
+                documentIdentityElements: [documentIdentityElement]
+            );
+
+            var sessionUpdateResult = await CreateUpdate()
+                .UpdateById(sessionUpdateRequest, Connection!, Transaction!, traceId);
+
+            sessionUpdateResult.Should().BeOfType<UpdateResult.UpdateSuccess>();
+
+            var getResult = await CreateGetById()
+                .GetById(
+                    CreateGetRequest("CourseOffering", _courseOfferingDocumentUuid),
+                    Connection!,
+                    Transaction!
+                );
+
+            getResult!.Should().BeOfType<GetResult.GetSuccess>();
+            (getResult! as GetResult.GetSuccess)!.DocumentUuid.Value.Should().Be(_courseOfferingDocumentUuid);
+            (getResult! as GetResult.GetSuccess)!.EdfiDoc.ToJsonString().Should().Contain("Fourth Quarter");
+
+            true.Should().BeTrue();
+        }
+    }
```

Comment:
How about an additional test that triggers the recursion?

Reply:
hah I didn't mean to leave this in. But yes on the recursion test.

Reply:
Updated the test. Previously it was just Session -> CourseOffering. Now it is Session -> CourseOffering -> Section (x2). This tests recursion (courseOffering -> section) as well as multiple same level.